### PR TITLE
RenderDocCmd.apk arguments, plus Internet permissions to use sockets.

### PR DESCRIPTION
**Pull request by michaelrgb**

### CODE REVIEW

File: renderdoccmd/renderdoccmd_android.cpp [hunk 1]
```diff
@@ -38,7 +38,7 @@ extern "C" {
 #include <android/log.h>
 #define LOGCAT_TAG "renderdoc"
 
-using std::string;
+using namespace std;
```
Comment (baldurk) [inline renderdoccmd/renderdoccmd_android.cpp]: I'd prefer not bringing in the entire namespace here, even if it is just an isolated cpp file. You should just be able to get away with vector, string and istringstream from what I can see.

File: renderdoccmd/renderdoccmd_android.cpp [hunk 2]
```diff
@@ -70,28 +70,62 @@ void DisplayRendererPreview(ReplayRenderer *renderer, TextureDisplay &displayCfg
...
+      if(!args.size())
+        break;    // Nothing for APK to do.
 
-      __android_log_print(ANDROID_LOG_INFO, LOGCAT_TAG,
-                          "APP_CMD_INIT_WINDOW: android_state->window: %p", android_state->window);
+      const char **argv = new const char *[args.size() + 1];
+      argv[0] = "renderdoccmd";
+      for(size_t i = 0; i < args.size(); i++)
+        argv[i + 1] = args[i].c_str();
+      renderdoccmd(args.size() + 1, (char **)argv);
```
Comment (baldurk) [inline renderdoccmd/renderdoccmd_android.cpp]: FYI the `renderdoccmd(int, char**)` just forwards to `renderdoccmd(std::vector<std::string>)` - see [renderdoccmd.cpp](https://github.com/baldurk/renderdoc/blob/master/renderdoccmd/renderdoccmd.cpp#L689). They're both declared in `renderdoccmd.h` so you can just `args.insert(args.begin(), "renderdoccmd")` and pass the existing vector if you prefer for slightly cleaner code.

**Comment by baldurk** [comment]
Aside from the notes this seems like it would merge just fine but the title says "merge manually" - is there something needing manual intervention?

**Comment by michaelrgb** [comment]
This was my way of saying that my detailed logs for each of my commits are already adequate, and it doesnt seem to serve any purpose to group them together given that the commits have different purposes. One is for network issues and one is for apk arguments, so shall I put something generic that covers both in the title? 

I will fix the namespace issue.

**Comment by baldurk** [comment]
Ah I see. That's fine - the PR title/merge commit message is mostly just so that it's vaguely possible to tell what the commits were, but mostly to link back here so you can see it in detail. As long as it is somewhat on topic I don't mind much what it is. I only queried it because I wasn't sure if there was something I was missing needing manual work.

**Pull request merged**